### ISSUE TITLE
[NgTemplateOutlet] `ngOutletContext` should be `NgTemplateOutletContext`

### ISSUE THREAD

**Issue by vicb** [angular/angular#11614]
All inputs of structural directives should be prefixed with the selector name so that the \* syntax could be used

`<ng-container *ngOutletContext></ng-container>`

fix:
- add a new input alias,
- deprecate `ngOutletContext`

/cc @pkozlowski-opensource

**Comment by pkozlowski-opensource** [comment]
@vicb while I agree on the general principle of supporting `*` sugar syntax everywhere, I'm not sure it is applicable to the `NgTemplateOutlet` directive. I might be reaching incorrect conclussions or misunderstanding your intention so let me share how I think about this directive.

First of all we should start by observing that currently the `<template>` tag is used to mean 2 radically different concepts:
* content capturing (everything you've got between `<template>` and `</template>` gets captured as a `TemplateRef`
* insertion point (`<template>` is just a way to get a `ViewContainerRef` on sth that doesn't render in the final DOM).

The `*` syntax makes only sense when `<template>` is as content capturing and not so much when marking content insertion point. The trouble today is that in most usages today (ngIf, ngFor etc.) we see `<template>` used as _both_ capturing mechanism and insertion point marker! So it is kind of hard to see that those concepts are really distinct. But if we consider `NgTemplateOutlet` this distinction is crucial! 

The typical usage always goes like follows:

```html
<template #foo>
  <span>I'm capturing this content</span>
</template>

<template [ngTemplateOutlet]="foo"></template> <!-- just marking an insertion point here--->
```

As I've mentioned previously I'm not sure the sugared syntax makes sense in the context of `ngTemplateOutlet` - if we try to use it we would get:

```html
<span *ngTemplateOutlet>I'm capturing this content</span>
```

which is really an equivalent... of static content (`<span>I'm capturing this content</span>`)!

tl;dr; I feel like I'm missing your intention here as currently I don't see how the sugared syntax could be useful for the `NgTemplateOutlet` directive...

**Comment by DzmitryShylovich** [comment]
@vicb can u explain your intention on this?

**Comment by pkozlowski-opensource** [comment]
This was changed in c0178de0e2d2162023ef81d2dbc358644ae37da8
